grp/classic.gd

```diff
@@ -227,19 +236,59 @@ DeclareConstructor( "GeneralOrthogonalGroupCons",
 ##  <#/GAPDoc>
 ##
 BindGlobal( "GeneralOrthogonalGroup", function ( arg )
-  if   Length( arg ) = 2 then
+  if   Length( arg ) = 1 then
+    # form (matrix, form object, or group with stored form)
+    return GeneralOrthogonalGroupCons( IsMatrixGroup, arg[1] );
+  elif Length( arg ) = 2 and IsInt( arg[1] ) then
+    # (d, q) or (d, R)
     return GeneralOrthogonalGroupCons( IsMatrixGroup, 0, arg[1], arg[2] );
   elif Length( arg ) = 3 and IsInt(arg[1]) and IsInt(arg[2]) and
     (IsInt(arg[3]) or IsRing(arg[3])) then
+    # (e, d, q) or (e, d, R)
     return GeneralOrthogonalGroupCons( IsMatrixGroup,arg[1],arg[2],arg[3] );
+  elif Length( arg ) = 3 and IsInt(arg[1]) and
+    (IsInt(arg[2]) or IsRing(arg[2])) and
+    (IsMatrixObj(arg[3]) or
+     (IsBoundGlobal( "IsQuadraticForm" ) and ValueGlobal( "IsQuadraticForm" )(arg[3])) or
+     HasInvariantQuadraticForm(arg[3])) then
+    # (d, q, form) or (d, R, form)
+    return GeneralOrthogonalGroupCons( IsMatrixGroup,0,arg[1],arg[2],arg[3] );
```

Comment:
Well, what should be the rule? The use of whitespace was not consistent before the changes proposed here, and it is not consistent afterwards. Should one really care about it?
(For the code line in question, one could argue that it would exceed 80 characters with the additional whitespace. For other lines, this argument would not apply.)